tests/run-make/c-link-to-rust-dylib/rmake.rs

```diff
@@ -0,0 +1,41 @@
+// This test checks that C linking with Rust does not encounter any errors, with dynamic libraries.
+// See <https://github.com/rust-lang/rust/issues/10434>.
+
+//@ ignore-cross-compile
+
+use std::fs::remove_file;
+
+use run_make_support::{
+    dynamic_lib_extension, is_msvc, read_dir, run, run_fail, rustc, tmp_dir, Cc,
+};
+
+fn main() {
+    rustc().input("foo.rs").run();
+
+    if is_msvc() {
+        Cc::new().input("bar.c").arg(tmp_dir().join("foo.dll.lib")).out_exe("bar").run();
```

Comment:
Oh, I think this isn't cleared because it doesn't start with `lib`? Might as well reset the whole `tmp_dir()` directory?

* Note that removing `tmp_dir` as a whole will fail, this is currently being addressed in https://github.com/rust-lang/rust/pull/125827

Reply:
Funny case. Well, adding a `fs::remove` then.